wandb/sdk/launch/agent/agent.py

```diff
@@ -36,6 +39,97 @@
 _logger = logging.getLogger(__name__)
 
 
+def init_pool_processes(jobs: Dict[str, Dict[str, str]], lock: threading.Lock) -> None:
+    global _jobs
+    global _jobs_lock
+    _jobs = jobs  # type: ignore
+    _jobs_lock = lock  # type: ignore
+
+
+def thread_run_job(
+    launch_spec: Dict[str, Any],
+    job: Dict[str, Any],
+    default_config: Dict[str, Any],
+    api: Api,
+    jobs_event: threading.Event,
+) -> None:
+    try:
+        _run_job(launch_spec, job, default_config, api, jobs_event)
+    except Exception:
+        wandb.termerror(f"Error running job: {traceback.format_exc()}")
+        api.ack_run_queue_item(job["runQueueItemId"])
+
+
+def _run_job(
```

Comment:
I honestly don't really understand the logic of pulling this out of the class. I understand that it **can**, but doesn't it kinda feel like a class method?

Reply:
fixed